src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -0,0 +1,359 @@
+         * which checks for one-touch feasibility.
+         *
+         * @param event ReceiverFSM::Update event
+         */
+        const auto update_onetouch = [this](auto event) {
+            auto best_shot = findFeasibleShot(event.common.world, event.common.robot);
+            auto one_touch = getOneTouchShotPositionAndOrientation(
+                event.common.robot, event.common.world.ball(),
+                best_shot->getPointToShootAt());
+
+            if (best_shot)
+            {
+                if (event.control_params.pass)
+                {
+                    event.common.set_intent(std::make_unique<MoveIntent>(
+                        event.common.robot.id(), one_touch.getPointToShootAt(),
+                        one_touch.getOpenAngle(), 0, DribblerMode::OFF,
+                        BallCollisionType::ALLOW,
+                        AutoChipOrKick{AutoChipOrKickMode::AUTOKICK,
+                                       BALL_MAX_SPEED_METERS_PER_SECOND},
+                        MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0,
+                        event.common.robot.robotConstants()));
+                }
+            }
+        };
+
+        /**
+         * One-touch shot is not possible, just receive ball as cleanly as possible.
+         *
+         * @param event ReceiverFSM::Update event
+         */
+        const auto update_receive = [this](auto event) {
+            if (event.control_params.pass)
+            {
+                event.common.set_intent(std::make_unique<MoveIntent>(
+                    event.common.robot.id(), event.control_params.pass->receiverPoint(),
+                    event.control_params.pass->receiverOrientation(), 0,
+                    DribblerMode::MAX_FORCE, BallCollisionType::ALLOW,
+                    AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                    MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0,
+                    event.common.robot.robotConstants()));
+            }
+        };
+
+        /**
+         * Constantly adjust the receives position to be directly
+         * infront of the ball for better reception. This is especially
+         * useful for long passes where the ball might not end up
+         * exactly at the pass.receiverPoint()
+         *
+         * @param event ReceiverFSM::Update event
+         */
+        const auto adjust_receive = [this](auto event) {
+            auto ball      = event.common.world.ball();
+            auto robot_pos = event.common.robot.position();
+
+            if ((ball.position() - robot_pos).length() >
+                DIST_TO_FRONT_OF_ROBOT_METERS + 2 * BALL_MAX_RADIUS_METERS)
+            {
+                Point ball_receive_pos = ball.position();
+
+                if (ball.velocity().length() > MIN_PASS_START_SPEED)
+                {
+                    ball_receive_pos = closestPoint(
+                        robot_pos,
+                        Line(ball.position(), ball.position() + ball.velocity()));
```

Comment:
if the ball goes past the robot but 1) Its not a stray pass and 2) its speed is > `MIN_PASS_START_START_SPEED` , will this cause the robot to wait for the ball to come to it?

Reply:
yeah if its not a stray pass, then we will try to receive cleanly

the "adjust_receive" should account for minor variations before the stray pass angle, and if its past the stray pass angle thats handled by the stray pass stuff

so i guess the only time we have a problem is if the stray pass detection doesn't overlap the `adjust_receive` states ability to adjust